server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+import sys
+import glob
+import errno
+import hashlib
+import shutil
+import tempfile
+from enum import Enum
+from argparse import ArgumentParser
+from s3backup import S3Config, Entry
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger, md5sum
+from pbench.report import Report
+
+
+_NAME_ = "pbench-audit-server"
+cnt=0
+
+def verify_subdirs(controller, cnt):
+
+    sub_dir = os.listdir(controller)
+
+    # Checking if the list is empty or not
+    if len(sub_dir) == 0:
+        print("\nThere is no state directory\n")
+    else:
+        for state in sub_dir:
+            if os.path.isfile(os.path.join(controller,state)):
+                cnt = cnt+1
+
+            if os.path.islink(os.path.join(controller,state)):
+                print("\nUnexpected Symlinks found in this state directory\n")
+                cnt=cnt+1
+
+            files = glob.iglob(os.path.join(controller,state,'*'))
+
+            for f in files:
+                if not f.endswith('.tar.xz') and not f.endswith('.tar.xz.md5'):
+                    print("\nFile with wrong Extension\n")
+
+    return cnt
+
+def verify_archive(archive):
```

Comment:
Good start here.  Can you see exactly how the various `find` commands work in the current audit code and compare?

Reply:
Yes sure